sway/tree/view.c

```diff
@@ -450,13 +450,27 @@ static struct sway_container *select_workspace(struct sway_view *view) {
 
 	// Check if there's any `assign` criteria for the view
 	list_t *criterias = criteria_for_view(view,
-			CT_ASSIGN_WORKSPACE | CT_ASSIGN_OUTPUT);
+			CT_ASSIGN_WORKSPACE | CT_ASSIGN_WORKSPACE_NUMBER | CT_ASSIGN_OUTPUT);
 	struct sway_container *ws = NULL;
 	for (int i = 0; i < criterias->length; ++i) {
 		struct criteria *criteria = criterias->items[i];
-		if (criteria->type == CT_ASSIGN_WORKSPACE) {
-			ws = workspace_by_name(criteria->target);
+		if (criteria->type == CT_ASSIGN_OUTPUT) {
+			// CT_ASSIGN_OUTPUT
```

Comment:
This comment was originally added because it was in the `else` block. If this code will stay here (see my other comments) then we should remove this comment and probably add a `// CT_ASSIGN_WORKSPACE` to the `else` block.